src/allmydata/test/test_istorageserver.py

```diff
@@ -248,12 +248,31 @@ def test_written_shares_are_readable(self):
             (yield buckets[2].callRemote("read", 0, 1024)), b"3" * 512 + b"4" * 512
         )
 
-    @skipIf(True, "https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3801")
+    @inlineCallbacks
     def test_overlapping_writes(self):
         """
-        The policy for overlapping writes is TBD:
-        https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3801
+        Overlapping, non-identical writes in immutable uploads fail.
```

Comment:
Should we have an interface test for overlapping identical writes too?

Reply:
OK.